Feature/9 setting base

5nam opened the pull request:
응답 관련 Base 생성했습니다.

1. BaseResponse - 성공 시 반환할 Response
2. BaseErrorResponse - 실패 시 반환할 Response
3. BaseException - 예외 처리 시에 작성해두고 이 값으로 BaseErrorResponse 생성 가능

이렇게 성공/실패로 나눠놓은 이유는 개발할 때 명시적으로 성공인지 실패인지 나누는 게 좋다고 생각했기 때문입니다.

5nam committed:
✨ 추가 BaseResponse
1. 필수로 들어가야 하는 필드들은 final 붙임
2. code, message, data 생성자 + code, message 만 넘겨주는 생성자

5nam committed:
✨ 추가 BaseErrorResponse, BaseException

1. BaseException 으로 예외처리 시 BaseException 생성
2. BaseErrorResponse 생성자 매개변수로 BaseException 넘겨서, BaseErrorResponse 생성
- 두 개를 생성한 이유는 BaseException 은 Exception 으로 error 를 주고, BaseErrorResponse 는 커스텀으로 줄 수 있도록 구분

5nam committed:
🩹 변경 @RequiredArgsConstructor 로 생성자 대체

5nam committed:
💡 삭제 불필요한 주석

g00hyun reviewed (approved):
필요한 부분은 다 있는것 같아서 승인 합니다 :)
response에 대해 정확한 확신이 없어서 지금으로썬 피드백 드릴만한게 없네요!
추후에 확신이 들면서 수정필요한 부분있으면 추가적으로 PR날릴게요 :)

5nam committed:
♻️ 생성 API 응답 통일
1. API 응답 통일 코드 BaseResponseDto 생성
: isSuccess, code, message, result 으로 구성
2. 오류 타입 통일 코드 ResponseCode 생성
: 에러 컨벤션은 논의 필요

5nam committed:
♻️ 추가 응답코드 & 에러핸들러
1. RestControllerAdvice 가 @RestController 가 붙은 대상에서 Exception 발생하는 것을 감지하여 Exception 핸들링

5nam committed:
🩹이동 MasterExceptionHandler 파일 위치

5nam committed:
:adhesive_bandage: 수정 BaseResponseDto Failure 응답 data 부분
1. Failure 응답 data 변수 -> null 로 고정하도록 변경
